File: app/src/main/java/com/wzf/wucarryme/modules/care/domain/BuySellORM.java
package com.wzf.wucarryme.modules.care.domain;


import com.litesuits.orm.db.annotation.Column;
import com.litesuits.orm.db.annotation.Ignore;
import com.litesuits.orm.db.annotation.NotNull;
import com.litesuits.orm.db.annotation.PrimaryKey;
import com.litesuits.orm.db.annotation.Table;
import com.litesuits.orm.db.enums.AssignType;

/**
 * @author wzf
 * @date 2018/3/22
 */
@Table("BUY_SELL")
public class BuySellORM {
    @Column("ID_")
    @PrimaryKey(AssignType.AUTO_INCREMENT)
    private int id;
    @NotNull
    @Column("BLOG_TIME")
    private String blogTime;
    @Column("CATEGORY_")
    private String category;
    @NotNull
    @Column("ACTION_")
    private String action;
    @Column("STOCK1_")
    private String stock1;
    @Column("STOCK2_")
    private String stock2;
    @Column("DESC_")
    private String desc;
    @Column("LOG_TIME")
    private String logTime;
    @Ignore
    private int count1;
    @Ignore
    private int count2;

    public BuySellORM(String blogTime, String category, String action, String stock1, String stock2, String desc,
                      String logTime) {
        this.blogTime = blogTime;
        this.category = category;
        this.action = action;
        this.stock1 = stock1;
        this.stock2 = stock2;
        this.desc = desc;
        this.logTime = logTime;
    }

    public BuySellORM(String blogTime, String logTime) {
        this.blogTime = blogTime;
        this.logTime = logTime;
    }

    public BuySellORM() {

    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getLogTime() {
        return logTime;
    }

    public void setLogTime(String logTime) {
        this.logTime = logTime;
    }

    public String getBlogTime() {
        return blogTime;
    }

    public void setBlogTime(String blogTime) {
        this.blogTime = blogTime;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getAction() {
        return action;
    }

    public void setAction(String action) {
        this.action = action;
    }

    public String getStock1() {
        return stock1;
    }

    public void setStock1(String stock1) {
        this.stock1 = stock1;
    }

    public String getStock2() {
        return stock2;
    }

    public void setStock2(String stock2) {
        this.stock2 = stock2;
    }

    public String getDesc() {
        return desc;
    }

    public void setDesc(String desc) {
        this.desc = desc;
    }

    public int getCount1() {
        return count1;
    }

    public void setCount1(int count1) {
        this.count1 = count1;
    }

    public int getCount2() {
        return count2;
    }

    public void setCount2(int count2) {
        this.count2 = count2;
    }
}

File: app/src/main/java/com/wzf/wucarryme/modules/care/domain/BuySellORM.kt
package com.wzf.wucarryme.modules.care.domain

import com.litesuits.orm.db.annotation.*
import com.litesuits.orm.db.enums.AssignType

/**
 * @author wzf
 * @date 2018/3/22
 */
@Table("BUY_SELL")
class BuySellORM {
    @Column("ID_")
    @PrimaryKey(AssignType.AUTO_INCREMENT)
    var id: Int = 0
    @NotNull
    @Column("BLOG_TIME")
    var blogTime: String? = null
    @Column("CATEGORY_")
    var category: String? = null
    @NotNull
    @Column("ACTION_")
    var action: String? = null
    @Column("STOCK1_")
    var stock1: String? = null
    @Column("STOCK2_")
    var stock2: String? = null
    @Column("STOCK1_PRICE")
    var stock1Price: String? = null
    @Column("STOCK2_PRICE")
    var stock2Price: String? = null
    @Column("STOCK1_RETURN")
    var stock1Return: String? = null
    @Column("STOCK2_RETURN")
    var stock2Return: String? = null
    @Column("DESC_")
    var desc: String? = null
    @Column("LOG_TIME")
    var logTime: String? = null
    @Ignore
    var count1: Int = 0
    @Ignore
    var count2: Int = 0

    constructor(blogTime: String, category: String, action: String, stock1: String, stock2: String, desc: String,
                logTime: String) {
        this.blogTime = blogTime
        this.category = category
        this.action = action
        this.stock1 = stock1
        this.stock2 = stock2
        this.desc = desc
        this.logTime = logTime
    }

    constructor(blogTime: String, logTime: String) {
        this.blogTime = blogTime
        this.logTime = logTime
    }

    constructor()

    override fun toString(): String {
        return "$blogTime [$action] [$category]<br/>" +
            "<table><tr><td>$count1</td><td>$stock1</td><td>$stock1Price</td><td>$stock1Return</td></tr>" +
            "<tr><td>$count2</td><td>$stock2</td><td>$stock2Price</td><td>$stock2Return</td></tr></table>" +
            "<br/>$desc<br/>$logTime"
    }
}
